akka-http2-support/src/main/scala/akka/http/impl/engine/http2/FrameParser.scala

```diff
@@ -66,7 +65,7 @@ class FrameParser(shouldReadPreface: Boolean) extends ByteStringParser[FrameEven
           if (priority) payload.readByte() & 0xff
           else 0
 
-        // TODO: check that streamId != 0
+        Http2Compliance.requirePositiveStreamId(streamId)
```

Comment:
Actually, that's not even correct: for incoming HEADERS from client it needs to be an odd streamId. We won't see even streamIds as long as we don't support sending out PUSH_PROMISEs.